### PR TITLE
Add a more specialized test for squash (UI)

### PR DESCRIPTION
**Pull request by mkondratek**
not a solution. only better testing for https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1079

### CODE REVIEW

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 1]
```diff
@@ -232,21 +232,65 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
...
+        "HOTFIX Add the trigger - fixes",
+        "Use new icons",
+        "Use new icons",
+        "allow-ownership-link",
+        "build-chain",
+        "call-ws",
+        "develop",
+        "hotfix/add-trigger",
+        "master",
+        "update-icons"
```
Comment (PawelLipski) [inline src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:260]: Hmmm this ordering is alphabetical atm, right? 🤔 would be good to fix #1334 first

Comment (PawelLipski) [reply src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:260]: So that the ordering tells more which commit belongs to which branch

**Comment by PawelLipski** [comment]
> java.lang.AssertionError: expected:

Hmm okay so you've managed to repro the #1079 bug locally right?

Anyway, lemme refrain from diagnosis until is #1334 is addressed (so that there's more clarity as to what's compared to what) 🤔

**Comment by mkondratek** [comment]
> > java.lang.AssertionError: expected:
> 
> Hmm okay so you've managed to repro the #1079 bug locally right?
> 
> Anyway, lemme refrain from diagnosis until is #1334 is addressed (so that there's more clarity as to what's compared to what) 🤔

no no no, my question is about `getCommitsUntilParent` method - why does is return `Develop commit` (and probably a more important question, why `isNonRoot` got true on the develop branch)?

File: src/uiTest/resources/project.rhino.js [hunk 2]
```diff
@@ -98,6 +98,23 @@ function Project(underlyingProject) {
...
   };
 
+  this.getManagedBranchesAndCommits = function () {
+    const snapshot = getGraphTable().getGitMacheteRepositorySnapshot();
+    if (snapshot != null) {
+      return snapshot.getManagedBranches().flatMap(b => {
+        const branchesAndCommits = new LinkedList();
+        if (b.isNonRoot()) {
+          const nonRoot = b.asNonRoot();
+          branchesAndCommits.addAll(nonRoot.getUniqueCommits().map(c => c.getShortMessage()).toJavaList());
```
Comment (mkondratek) [inline src/uiTest/resources/project.rhino.js:108]: it's here exactly - using `getUniqueCommits` works as expected (at least to me), but with `getCommitsUntilParent` the `Develop commit` is included in the result. 

Probably it's not from the develop branch (`isNonRoot` for develop works fine) but some other branch decided that the parent is root (not develop as it should) 🤔

Comment (PawelLipski) [reply src/uiTest/resources/project.rhino.js:108]: Wow okay.... weird 😯

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**